server/rootvalue/rootvalue.go

```diff
@@ -0,0 +1,723 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rootvalue
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb/durable"
+	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
+	"github.com/dolthub/dolt/go/store/hash"
+	"github.com/dolthub/dolt/go/store/prolly/tree"
+	"github.com/dolthub/dolt/go/store/types"
+)
+
+const (
+	ddbRootStructName = "dolt_db_root"
+	tablesKey         = "tables"
+	foreignKeyKey     = "foreign_key"
+	featureVersKey    = "feature_ver"
+	rootCollationKey  = "root_collation_key"
+)
+
+// DoltgresFeatureVersion is Doltgres' feature version. We use Dolt's feature version added to our own.
+var DoltgresFeatureVersion = doltdb.DoltFeatureVersion + 0
+
+// rootValue is Dolt's implementation of RootValue.
+type rootValue struct {
```

Comment:
Lots of duplication here. What if we embedded dolt's root value implementation instead?